vfs/s3fs.go

```diff
@@ -138,51 +137,17 @@ func (fs *S3Fs) Stat(name string) (os.FileInfo, error) {
 	}
 	// the requested file could still be a directory as a zero bytes key
 	// with a forwarding slash.
-	// As last resort we do a list dir to find it
-	return fs.getStatCompat(name)
-}
-
-func (fs *S3Fs) getStatCompat(name string) (os.FileInfo, error) {
-	var result FileInfo
-	prefix := path.Dir(name)
-	if prefix == "/" || prefix == "." {
-		prefix = ""
-	} else {
-		prefix = strings.TrimPrefix(prefix, "/")
-		if !strings.HasSuffix(prefix, "/") {
-			prefix += "/"
-		}
+	obj, err = fs.headObject(strings.TrimPrefix(name, "/") + "/")
+	if err == nil {
+		objSize := *obj.ContentLength
+		objectModTime := *obj.LastModified
+		isDir := *obj.ContentType == dirMimeType
```

Comment:
if we get here we created a dir with an old SFTPGo version or using s3fs and the content type is not dirMimeType, we should add this check for the first stat, here we should always set isDir to true, sorry if I was not clear enough